asllib/doc/Statements.tex

```diff
@@ -3098,5 +3098,61 @@ \subsection{Syntax}
 \end{flalign*}
 
 \subsection{Abstract Syntax}
-\subsection{Typing}
+\begin{flalign*}
+\stmt \derives\ & \SPragma(\Tidentifier, \overtext{\expr^*}{args}) &
+\end{flalign*}
+
+\subsubsection{ASTRule.SPragma}
+\begin{mathpar}
+\inferrule{
+  \buildclist[\Nexpr](\vargs) \astarrow \astversion{\vargs}
+}{
+    \buildstmt(\overname{\Nstmt(\Tpragma, \Tidentifier(\id), \namednode{\vargs}{\Clist{\Nexpr}}, \Tsemicolon)}{\vparsednode})
+  \astarrow
+    \overname{\SPragma(\id, \astversion{\vargs})}{\vastnode}
+}
+\end{mathpar}
+
+\subsubsection{TypingRule.SPragma\label{sec:TypingRule.SPragma}}
+\subsubsection{Prose}
+All of the following apply:
+\begin{itemize}
+  \item $\vs$ is a pragma statement, that is, $\SPragma$;
+  \item applying $\annotateexprlist$ to annotate the expression list $\vargs$ in $\tenv$ yields \\
+        $\vargsp$\ProseOrTypeError;
+  \item $\news$ is the pragma using $\vargsp$, that is, $\SPragma(\id, \vargsp)$;
+  \item $\newtenv$ is $\tenv$.
+\end{itemize}
+
+\subsubsection{Formally}
+\begin{mathpar}
+\inferrule{
+  \annotateexprlist(\tenv, \vargs) \typearrow \vargsp \OrTypeError
+}{
+  \annotatestmt(\tenv, \overname{\SPragma(\id, \vargs)}{\vs}) \typearrow (\overname{\SPragma(\id, \vargsp)}{\news}, \tenv)
+}
+\end{mathpar}
+\CodeSubsection{\SPragmaBegin}{\SPragmaEnd}{../Typing.ml}
+
 \subsection{Semantics}
```

Comment:
I think you can remove `SemanticsRUle.SPramga` and write `Pragmas are not associated with a semantics.`